### PR TITLE
Expand tests related to task system

### PR DESCRIPTION
**Pull request by kersommoura**
Add test to cancel tasks, to delete a running task, and to delete a
nonexistent task.

closes: PulpQE/pulp-smash#976
https://pulp.plan.io/issues/3634
closes:#3634

### CODE REVIEW

**Comment by kersommoura** [comment]
@rochacbruno. CI is not happy about code/coverage the tests did not mess with any of those.
Any other comment?

File: pulpcore/tests/functional/api/using_plugin/test_tasking.py [hunk 1]
```diff
@@ -61,3 +65,68 @@ def test_all(self):
...
+        delete_orphans(self.cfg)
+
+        repo = self.client.post(REPO_PATH, gen_repo())
+        self.addCleanup(self.client.delete, repo['_href'])
+
+        body = gen_remote(url=FILE_LARGE_FIXTURE_MANIFEST_URL)
+        remote = self.client.post(FILE_REMOTE_PATH, body)
+        self.addCleanup(self.client.delete, remote['_href'])
+
+        client = api.Client(self.cfg, api.code_handler)
```
Comment (rochacbruno) [inline pulpcore/tests/functional/api/using_plugin/test_tasking.py]:
```suggestion
       
```

File: pulpcore/tests/functional/api/using_plugin/test_tasking.py [hunk 2]
```diff
@@ -61,3 +65,68 @@ def test_all(self):
...
+        return client.post(
```
Comment (rochacbruno) [inline pulpcore/tests/functional/api/using_plugin/test_tasking.py]:
```suggestion
        return self.client.using_handler(api.code_handler).post(
```

File: pulpcore/tests/functional/api/using_plugin/test_tasking.py [hunk 3]
```diff
@@ -61,3 +65,68 @@ def test_all(self):
...
+            urljoin(remote['_href'], 'sync/'), {'repository': repo['_href']}
+        ).json()
+
+    @staticmethod
```
Comment (rochacbruno) [inline pulpcore/tests/functional/api/using_plugin/test_tasking.py]: Why a static method here?

If you have a regular method you can reuse the same client (with using_handler) and the same cfg already present in the class.

**Comment by kersommoura** [comment]
Updating this one now.

**Comment by kersommoura** [comment]
@rochacbruno, updated per comments.

**Pull request merged**